### PR TITLE
Added counter for input fields with maximum-length attribute

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Adds js-snippet to ...

- admin product edit page
- admin product attributes edit page
- admin config section

... to show how many chars are left for valid input.

__Note:__ requires `validate-length` and `maximum-length-XXX` attribute

counter

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by justinbeaty** [comment]
Nice quality of life feature! My only suggestion is to style it like this, which I think is more standard UI.

image


```js
        // display the value, run once at load and on each observed keyup
        countdown: function(element) {
            var elm = $(element);
            var curLen = elm.getValue().length;
            var maxLen = elm.retrieve('maxLen');
            var count  = maxLen - curLen;
            var counter = elm.retrieve('counter');
            counter.update(' (' + curLen + '/' + maxLen + ')');
            if (curLen > maxLen) {
                counter.setStyle({'color': 'red'});
            } else {
                counter.setStyle({'color': 'inherit'});
            }
            return elm;
        }
```

I also changed the conditional to just `>` instead of `>=` because I was allowed to save an attribute with code with exactly 30 characters, despite the note: "Maximum length of attribute code must be less then 30 symbols" (unrelated, there's also a typo there, should be "than").

**Review by justinbeaty — APPROVED** [review]

File: js/mage/adminhtml/input-counter.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                counter.setStyle({'color': 'red'});
+            } else {
+                counter.setStyle({'color': 'inherit'});
+            }
+            return elm;
+        }
+    });
+
+    // run setup and call countdown once outside of listener to initialize
+    $$('.validate-length').invoke('prepare_for_countdown').invoke('countdown');
```
Comment (fballiano) [inline js/mage/adminhtml/input-counter.js:35]: can we not use prototype?

Comment (sreichel) [reply js/mage/adminhtml/input-counter.js:35]: Of course, but I dont want to rewrite my 6y old code ;)

Comment (fballiano) [reply js/mage/adminhtml/input-counter.js:35]: I thought it was a couple of lines of prototypejs but actually it's all of it 😅

Comment (sreichel) [reply js/mage/adminhtml/input-counter.js:35]: My only argument is "it works" ... and it might be helpful.

**Review by fballiano — APPROVED** [review]

File: js/mage/adminhtml/input-counter.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// https://stackoverflow.com/a/44436408/5703627
+document.observe('dom:loaded', function() {
+    Element.addMethods({
+        // setup once, memoize the counter element and maxLen
```
Comment (kiatng) [inline js/mage/adminhtml/input-counter.js:4]: Typo:
```suggestion
        // setup once, memorize the counter element and maxLen
```

Comment (sreichel) [reply js/mage/adminhtml/input-counter.js]: Once again "thanks".

**Review by kiatng — APPROVED** [review]

**Review by fballiano — APPROVED** [review]

**Comment by fballiano** [comment]
merged and v20ed

**Comment by luigifab** [comment]
This create a bug: `document.observe('dom:loaded',`
This is a prototype way, then if you merge and minify all js files in one, this will crash:
  `Uncaught TypeError: document.observe is not a function`

**Comment by fballiano** [comment]
@luigifab if you change that line with `Event.observe(window, 'load',  function() {` does it work?

**Comment by sreichel** [comment]
Changing to `Event.observe(window, 'load',  function() {` works for me w/ merged JS files too.

@luigifab PR ready if confirmed.

**Comment by luigifab** [comment]
No, but with `window.addEventListener('load',  function() {` it's ok, and it works on new attribute product page.

**Pull request merged**